src/openscad.cc

```diff
@@ -429,10 +429,11 @@ int cmdline(const char *deps_output_file, const std::string &filename, const cha
 			root_geom = geomevaluator.evaluateGeometry(*tree.root(), true);
 			if (!root_geom) root_geom.reset(new CGAL_Nef_polyhedron());
 			if (viewOptions.renderer == RenderType::CGAL && root_geom->getDimension() == 3) {
-				auto N = dynamic_cast<const CGAL_Nef_polyhedron*>(root_geom.get());
-				if (!N) {
-					N = CGALUtils::createNefPolyhedronFromGeometry(*root_geom);
-					root_geom.reset(N);
+				if (!dynamic_pointer_cast<const CGAL_Nef_polyhedron>(root_geom) &&
+						!dynamic_pointer_cast<const GeometryList>(root_geom)) {
+					// FIXME: We should probably recursively convert all leaf nodes in the list
+					// to Nef polyhedron
```

Comment:
We may want to deal with this FIXME

Reply:
OK, I changed this section.  It was a little tricky since nested geometries are const, but I ended up calling `GeometryList::flatten` first (you had implemented it in the original code, but it wasn't yet used anywhere), then looping through that to convert and reset shared_ptrs where necessary.  I don't 100% know the reason to force all CGAL Nef, but if we are doing this, should the conversion maybe be moved into `GeometryEvaluator::evaluateGeometry` instead so that export from GUI works the same?  Maybe even cache the converted result?

Reply:
This was an  old comment, but I think the  rationale was that since the cmd-line arguments specifically asks for a CGAL evaluation, we should probably do that. `viewOptions.renderer == RenderType::CGAL`

Reply:
..and that option doesn't exist in the GUI

Reply:
ok, makes sense